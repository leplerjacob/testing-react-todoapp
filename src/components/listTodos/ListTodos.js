import React from "react";

function ListTodos(props) {
  let list = props.todos;

  return (
    <div>
      <ul>
        {list.map((item) => {
          return <li key={item.id}>{item.todo}</li>;
        })}
      </ul>
    </div>
  );
}

export default ListTodos;
