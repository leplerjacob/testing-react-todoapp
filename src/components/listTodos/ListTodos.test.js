import React from "react";
import ListTodos from "./ListTodos";
import Enzyme, { shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJSON from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

describe("ListTodos properties", () => {
  let wrapper;
  beforeEach(() => {
    const todos = [
      { id: 1, todo: "Walk the dog", date: "2021-10-20" },
      { id: 2, todo: "Visit grandparents", date: "2021-10-22" },
      { id: 3, todo: "Attend React conference", date: "2021-10-26" },
    ];
    wrapper = shallow(<ListTodos todos={todos} />);
  });
  afterEach(() => {
    wrapper.unmount();
  });
  test("renders ListTodos component", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  test("displays correct number of data passed as prop", () => {
    expect(wrapper.find("ul").children()).toHaveLength(3);
  });
  test("shows text of todo item", () => {
    expect(wrapper.find("ul").children("li").at(0).text()).toBe("Walk the dog");
    expect(wrapper.find("ul").children("li").at(1).text()).toBe(
      "Visit grandparents"
    );
    expect(wrapper.find("ul").children("li").at(2).text()).toBe(
      "Attend React conference"
    );
  });

});
