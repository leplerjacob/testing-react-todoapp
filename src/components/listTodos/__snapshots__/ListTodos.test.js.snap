// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTodos properties renders ListTodos component 1`] = `
<div>
  <ul>
    <li
      key="1"
    >
      Walk the dog
    </li>
    <li
      key="2"
    >
      Visit grandparents
    </li>
    <li
      key="3"
    >
      Attend React conference
    </li>
  </ul>
</div>
`;
